.ci/Jenkinsfile

```diff
@@ -330,45 +313,38 @@ stage('Build') {
         isMergeCommit = false
     }
 
-    pytorchDockerBuildMatrix.each { entry ->
-        // Extract the values from the buildMatrix
-        String kanikoCommand = entry[0]  // kanikoCommand is the command to run
-        String destinations = entry[1]  // The destinations to append to the kanikoCommand to push the image
-        String stagingImage = entry[2]  // stagingImage is where the built docker image is always pushed
-        // buildArgs contains the entry from the build matrix. It has the format [{key: key, value: value}, ...].
-        def buildArgs = entry[3]
+    pytorchDockerBuildMatrix.each { buildArgs ->
         Boolean isComposerImage = buildArgs['TARGET'] == 'composer_stage'
-
-            // If changing docker, build the docker images first
-            // Then, run pytest in the newly-built image
-
         Boolean shouldPushToDestinations = isMergeCommit && gitBranch == baseBranch
+        String label = buildArgs['TAGS'][0]
 
-        if (shouldPushToDestinations) {
-            kanikoCommand = "$kanikoCommand $destinations"
+        Boolean shouldBuildImage = didDockerChange  // Always build the image if the ./docker folder changed
+
+        if (!shouldPushToDestinations) {
+            buildArgs['TAGS'] = []
         }
 
-        if (isComposerImage) {
-            if (!buildArgs['COMPOSER_INSTALL_COMMAND'].contains('==')) {
-                // If this is a composer image and it is the latest tag, skip it -- we'll build the versioned tag instead
-                return
-            }
+        if (isComposerImage && buildArgs['COMPOSER_INSTALL_COMMAND'] == 'mosaicml[all]GIT_COMMIT') {
+            // If this is a composer image and it is the 'GIT_COMMIT' entry in the matrix
+            // then build and publish the image on the staging repo
             if (!isMergeCommit) {
                 // Skip composer images except on merge commits -- it's slow!
-                return
+                // return  // TODO (ravi) -- uncomment this line
```

Comment:
TODO
```suggestion
                return
```

Reply:
I'm going to approve to unblock you in the morning but please resolve this before merging.

Reply:
Removed this TODO